### PR TITLE
feat: add integration documentation

### PR DESCRIPTION
**Pull request by jrwbabylonlab**

### CODE REVIEW

File: bbn-1/integration/staking-backend.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+### Staking Transactions
+
+The system defines three types of transactions:
+- *Staking* which creates new stake by locking BTC in the self-custodial
+  Bitcoin Staking contract,
+- *Unbonding* which unbonds stake from the Bitcoin staking contract before the locking expires, and
+- *Withdraw* which extracts unlocked/unbonded stake to any wallet address specified by the staker.
+
+The full spec of the staking transactions can be found
+[here](https://github.com/babylonlabs-io/babylon/blob/a8c9d27ab1d489eb55c23cbb2c75b87e1a85afdb/docs/staking-script.md).
```
Comment (gitferry) [inline bbn-1/integration/staking-backend.md:36]:
```suggestion
[here](https://github.com/babylonlabs-io/babylon/releases/tag/v0.9.1).
```
Guess we can use a tagged version

File: bbn-1/integration/staking-backend.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+(described later) which collects their signatures before forwarding the fully
+signed transaction to Bitcoin.
+Note that the unbonding transaction requires the staker's signature, so the
+covenant committee cannot consume the staking transaction
+without the staker's approval.
+
+The list of covenant emulation committee members is a global parameter, and the
+signer servers are shared among all instances of the staking back-ends.
+
+## Back-End Staking System Components
```
Comment (gitferry) [inline bbn-1/integration/staking-backend.md:73]:
```suggestion
## Backend Staking System Components
```

File: bbn-1/integration/wallet.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+**Option-2**, host your own Bitcoin staking website that
+connects to your extension wallet that retrieves staking
+information from a back-end you operate.
+- To integrate with a staking website that uses the Babylon reference staking
+  dApp [implementation](https://github.com/babylonlabs-io/simple-staking), please
+  check the interface your wallet (or a wrapper of it) should adhere to 
+  [here](https://github.com/babylonlabs-io/simple-staking/blob/main/docs/WalletIntegration.md).
+- For information about developing your own Bitcoin staking website, please check
+    - our [TypeScript](https://github.com/babylonlabs-io/btc-staking-ts)
+      and [Golang](https://github.com/babylonlabs-io/babylon/tree/dev/btcstaking)
```
Comment (gitferry) [inline bbn-1/integration/wallet.md:46]:
```suggestion
      and [Golang](https://github.com/babylonlabs-io/babylon/tree/main/btcstaking/identifiable_staking.go)
```
Let's point to main and `identifiable_staking.go` to be more precise

File: bbn-1/integration/wallet.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      Bitcoin Staking libraries.
+    - our reference web application [implementation](https://github.com/babylonlabs-io/simple-staking).
+- For information about operating your own back-end, please read this
+  [document](./staking-backend.md)
+
+**Option-3**, develop Bitcoin staking as a feature of your extension wallet,
+which connects to either third party APIs
+(such as the Babylon [API](https://staking-api.testnet.babylonchain.io/swagger/index.html)) or a back-end you operate.
```
Comment (gitferry) [inline bbn-1/integration/wallet.md:54]:
```suggestion
(such as the Babylon [API](https://staking-api.testnet.babylonchain.io/swagger/index.html)) or a back-end you operate.
```

Would it be changed to mainnet?

File: bbn-1/integration/wallet.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+- For information about developing your own Bitcoin staking as a feature, please check
+    - our [TypeScript](https://github.com/babylonlabs-io/btc-staking-ts)
+      and [Golang](https://github.com/babylonlabs-io/babylon/tree/dev/btcstaking)
```
Comment (gitferry) [inline bbn-1/integration/wallet.md:57]:
```suggestion
      and [Golang](https://github.com/babylonlabs-io/babylon/tree/main/btcstaking/identifiable_staking.go)
```
same here

File: bbn-1/integration/wallet.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+**Option-2**, host your own Bitcoin staking website that connects to your
+extension wallet that retrieves staking information from a back-end you
+operate. Then embed your own Bitcoin staking website to your mobile app wallet.
+- To integrate with a staking website that uses the Babylon reference staking
+  dApp [implementation](https://github.com/babylonlabs-io/simple-staking), please
+  check the interface your mobile wallet (or a wrapper of it) should adhere to 
+  [here](https://github.com/babylonlabs-io/simple-staking/blob/main/docs/WalletIntegration.md).
+- For information about developing your own Bitcoin staking website, please check
+    - our [TypeScript](https://github.com/babylonlabs-io/btc-staking-ts)
+      and [Golang](https://github.com/babylonlabs-io/babylon/tree/dev/btcstaking)
```
Comment (gitferry) [inline bbn-1/integration/wallet.md:81]:
```suggestion
      and [Golang](https://github.com/babylonlabs-io/babylon/tree/main/btcstaking/identifiable_staking.go)
```

File: bbn-1/integration/wallet.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    - our reference web application [implementation](https://github.com/babylonlabs-io/simple-staking).
+- For information about operating your own back-end, please read this
+  [document](./staking-backend.md)
+
+**Option-3**, develop Bitcoin staking as a feature of your extension wallet,
+which connects to either third party APIs
+(such as the Babylon [API](https://staking-api.testnet.babylonchain.io/swagger/index.html)) or a back-end you operate.
+- For information about developing Bitcoin staking as a feature, please check
+    - our [TypeScript](https://github.com/babylonlabs-io/btc-staking-ts)
+      and [Golang](https://github.com/babylonlabs-io/babylon/tree/dev/btcstaking)
```
Comment (gitferry) [inline bbn-1/integration/wallet.md:92]:
```suggestion
      and [Golang](https://github.com/babylonlabs-io/babylon/tree/main/btcstaking/identifiable_staking.go)
```

File: bbn-1/integration/wallet.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  [document](./staking-backend.md)
+
+### 3. Hardware Wallet
+
+**Option-1**, develop Bitcoin staking as a feature of your extension wallet,
+which connects to either third party APIs
+(such as the Babylon [API](https://staking-api.testnet.babylonchain.io/swagger/index.html)) or a back-end you operate.
+- For information about developing your own Bitcoin staking website, please check
+    - our [TypeScript](https://github.com/babylonlabs-io/btc-staking-ts)
+      and [Golang](https://github.com/babylonlabs-io/babylon/tree/dev/btcstaking)
```
Comment (gitferry) [inline bbn-1/integration/wallet.md:105]:
```suggestion
      and [Golang](https://github.com/babylonlabs-io/babylon/tree/main/btcstaking/identifiable_staking.go)
```

**Review by gitferry — APPROVED** [review]
Currently all the api urls pointing to testnet under babylonchain. Do we want to change that later? Other than that, lgtm!

File: bbn-1/integration/wallet.md (hunk 4 above)
Comment (vitsalis) [reply bbn-1/integration/wallet.md:54]: We have it now

File: bbn-1/integration/staking-backend.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
+# Operating a Bitcoin Staking Backend
+
+In this document we describe a reference tech stack employed by Babylon for a
+back-end service that collects information about the babylon staking system from
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:4]:
```suggestion
back-end service that collects information about the Babylon staking system from
```

File: bbn-1/integration/staking-backend.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+Bitcoin and adhere to the staking parameters are considered as valid staking
+transactions.
+
+### Staking Transactions
+
+The system defines three types of transactions:
+- *Staking* which creates new stake by locking BTC in the self-custodial
+  Bitcoin Staking contract,
+- *Unbonding* which unbonds stake from the Bitcoin staking contract before the locking expires, and
+- *Withdraw* which extracts unlocked/unbonded stake to any wallet address specified by the staker.
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:33]:
```suggestion
- *Withdraw* which extracts unlocked/unbonded stake to the staker's address.
```

File: bbn-1/integration/staking-backend.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+The full spec of the staking transactions can be found
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:35]:
```suggestion
The full spec of the transactions involved in the Bitcoin staking protocol can be found
```

File: bbn-1/integration/staking-backend.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+
+Participants of the system can create staking transactions either through the
+staking dApp or the staker CLI:
+- The [Staking dApp](https://github.com/babylonlabs-io/simple-staking/tree/main)
+  is a user-friendly application allowing users to create, sign, and submit
+  staking transactions to the Bitcoin ledger.
+  It can either be built directly inside the Bitcoin wallet or can connect
+  to a wallet. It communicates with a back-end service (described later) that
+  has access to an unbonding pipeline in order to submit unbonding requests.
+- The [Staker CLI](https://github.com/babylonlabs-io/btc-staker/tree/main)
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:46]: Let's point to v0.5.x

File: bbn-1/integration/staking-backend.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+  through a wallet of their choice and submitting them to the Bitcoin ledger or
+  the Staking API service in the case of unbonding.
+
+One could also build their own staking application and interact with the staking backend.
+
+### Covenant Emulation Committee
+
+The covenant emulation committee is a set of entities responsible for approving
+on-demand unbonding requests. The members of the committee operate a
+[covenant signer](https://github.com/babylonlabs-io/covenant-signer/tree/main)
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:59]: Let's use v0.2.x

File: bbn-1/integration/staking-backend.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+If the requests are valid, then the signature is returned in the response.
+The covenant signer servers are contacted by a back-end unbonding pipeline
+(described later) which collects their signatures before forwarding the fully
+signed transaction to Bitcoin.
+Note that the unbonding transaction requires the staker's signature, so the
+covenant committee cannot consume the staking transaction
+without the staker's approval.
+
+The list of covenant emulation committee members is a global parameter, and the
+signer servers are shared among all instances of the staking back-ends.
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:71]:
```suggestion
signer servers can be accessed from all instances of the staking back-ends.
```

File: bbn-1/integration/wallet.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+    - our reference web application [implementation](https://github.com/babylonlabs-io/simple-staking).
+- For information about operating your own back-end, please read this
+  [document](./staking-backend.md)
+
+**Option-3**, develop Bitcoin staking as a feature of your extension wallet,
+which connects to either third party APIs
+(such as the Babylon [API](https://staking-api.testnet.babylonchain.io/swagger/index.html)) or a back-end you operate.
+- For information about developing your own Bitcoin staking as a feature, please check
+    - our [TypeScript](https://github.com/babylonlabs-io/btc-staking-ts/tree/main)
+      and [Golang](https://github.com/babylonlabs-io/babylon/tree/main/btcstaking/identifiable_staking.go)
```
Comment (vitsalis) [inline bbn-1/integration/wallet.md:57]: Let's version the above links -- as well as others in this guide

**Review by vitsalis — APPROVED** [review]
lgtm minor comments

**Review by gitferry — APPROVED** [review]

File: bbn-1/integration/staking-backend.md [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+[global parameters](../parameters) that specify what constitutes a valid
+staking transaction that the system recognizes. Transactions that are on
+Bitcoin and adhere to the staking parameters are considered as valid staking
+transactions.
+
+### Staking Transactions
+
+The system defines three types of transactions:
+- *Staking* which creates new stake by locking BTC in the self-custodial
+  Bitcoin Staking contract,
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:31]:
```suggestion
  Bitcoin Staking script,
```

File: bbn-1/integration/staking-backend.md [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+- *Unbonding* which unbonds stake from the Bitcoin staking contract before the locking expires, and
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:32]:
```suggestion
- *Unbonding* which unbonds stake from the Bitcoin staking script before the locking expires, and
```

File: bbn-1/integration/staking-backend.md [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+- *Withdraw* which extracts unlocked/unbonded stake to the staker's address.
+
+The full spec of the transactions involved in the Bitcoin staking protocol can be found
+[here](https://github.com/babylonlabs-io/babylon/blob/main/docs/staking-script.md).
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:36]:
```suggestion
[here](https://github.com/babylonlabs-io/babylon/blob/v0.9.x/docs/staking-script.md).
```

File: bbn-1/integration/staking-backend.md [hunk 19]
```diff
@@ -0,0 +1,125 @@
...
+Note that the unbonding transaction requires the staker's signature, so the
+covenant committee cannot consume the staking transaction
+without the staker's approval.
+
+The list of covenant emulation committee members is a global parameter, and the
+signer servers can be accessed from all instances of the staking back-ends.
+
+## BackEnd Staking System Components
+
+### [Staking Indexer](https://github.com/babylonlabs-io/staking-indexer)
```
Comment (vitsalis) [inline bbn-1/integration/staking-backend.md:75]: Let's point to the `v0.{minor}.x` branches in those links?

File: bbn-1/integration/wallet.md [hunk 20]
```diff
@@ -0,0 +1,115 @@
+# Wallet Integration
+
+The `bbn-1` will focus on the security of the staked Bitcoins by
```
Comment (vitsalis) [inline bbn-1/integration/wallet.md:3]:
```suggestion
The `bbn-1` network will focus on the security of the staked Bitcoins by
```

File: bbn-1/integration/wallet.md [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+exposing compatible APIs.
+To integrate with the Babylon hosted staking website, please check the
+interface your wallet (or a wrapper of it) should adhere to
+[here](https://github.com/babylonlabs-io/simple-staking/blob/main/docs/WalletIntegration.md).
+
+**Option-2**, host your own Bitcoin staking website that
+connects to your extension wallet that retrieves staking
+information from a back-end you operate.
+- To integrate with a staking website that uses the Babylon reference staking
+  dApp [implementation](https://github.com/babylonlabs-io/simple-staking), please
```
Comment (vitsalis) [inline bbn-1/integration/wallet.md:41]:
```suggestion
  dApp [implementation](https://github.com/babylonlabs-io/simple-staking/tree/main), please
```

Same for all other instances

File: bbn-1/integration/wallet.md [hunk 22]
```diff
@@ -0,0 +1,115 @@
...
+  check the interface your wallet (or a wrapper of it) should adhere to 
+  [here](https://github.com/babylonlabs-io/simple-staking/blob/main/docs/WalletIntegration.md).
+- For information about developing your own Bitcoin staking website, please check
+    - our [TypeScript](https://github.com/babylonlabs-io/btc-staking-ts)
```
Comment (vitsalis) [inline bbn-1/integration/wallet.md:45]:
```suggestion
    - our [TypeScript](https://github.com/babylonlabs-io/btc-staking-ts/tree/main)
```

Same for all other instances

**Review by vitsalis — APPROVED** [review]
Mostly recommendations on adding versions on links. While these will need to be updated with new minor version updates, I believe we should have control on what versions are listed in our docs as we are responsible for ensuring compatibility within them.

**Review by vitsalis — APPROVED** [review]

**Pull request merged**